NitroxModel/DataStructures/GameLogic/Entity.cs

```diff
@@ -118,5 +119,58 @@ private void JsonAfterDeserialization(StreamingContext context)
         {
             ProtoAfterDeserialization();
         }
+
+        public bool Equals(Entity other)
+        {
+            if (other is null)
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return SpawnedByServer == other.SpawnedByServer &&
+                   Equals(Transform, other.Transform) &&
+                   Equals(TechType, other.TechType) &&
+                   Equals(Id, other.Id) &&
+                   Level == other.Level &&
+                   ClassId == other.ClassId &&
+                   Equals(WaterParkId, other.WaterParkId) &&
+                   Equals(SerializedGameObject, other.SerializedGameObject) &&
+                   ExistsInGlobalRoot == other.ExistsInGlobalRoot &&
+                   Equals(ParentId, other.ParentId) &&
+                   Equals(Metadata, other.Metadata) &&
+                   ExistingGameObjectChildIndex == other.ExistingGameObjectChildIndex &&
+                   ChildEntities.SequenceEqual(other.ChildEntities);
+        }
+
+        public override bool Equals(object obj)
+        {
+            return ReferenceEquals(this, obj) || obj is Entity other && Equals(other);
+        }
+
+        public override int GetHashCode()
```

Comment:
Hmm this might be dangerous as we are indexing on entity a lot of places and using the default reference equals implementation.  Also, not sure what the performance characteristics of these are as they run on child entities.  Is there a way we can forgot messing with these?

Reply:
Yes I can use a custom equality checker in the EntityManager from NitroxServer.